### PR TITLE
fix time_nanosleep return type signature

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/8097

### CODE REVIEW

File: resources/functionMap.php [hunk 1]
```diff
@@ -12308,7 +12308,7 @@
 'tidyNode::isPhp' => ['bool'],
 'tidyNode::isText' => ['bool'],
 'time' => ['positive-int'],
-'time_nanosleep' => ['array{0:0|positive-int,1:0|positive-int}|bool', 'seconds'=>'int', 'nanoseconds'=>'int'],
+'time_nanosleep' => ['array{seconds:0|positive-int,nanoseconds:0|positive-int}|bool', 'seconds'=>'int', 'nanoseconds'=>'int'],
```
Comment (mad-briller) [inline resources/functionMap.php:12311]: would an int-range `int<0, max>` be better than unioning a constant int and `positive-int`?

Comment (staabm) [reply resources/functionMap.php:12311]: I don't think any of the two is "better" - they are equivalent.

Comment (mad-briller) [reply resources/functionMap.php:12311]: isn't a single `IntegerRangeType` preferable to a `UnionType` containing two different types with regards to performance?
semantically they are they same i agree, but will phpstan simplify it to reduce the php object count from `3` to `1` ?

Comment (rvanvelzen) [reply resources/functionMap.php:12311]: `0|positive-int` will be normalized to `int<0, max>` so in that sense they're equivalent

Comment (staabm) [reply resources/functionMap.php:12311]: if you find a example in which we find a measurable difference we should have a closer look.
as we don't have one, I would assume its a micro difference

Comment (mad-briller) [reply resources/functionMap.php:12311]: @rvanvelzen thank you, i'm still learning lots about phpstan's type system so forgive any silly questions i may have!

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**